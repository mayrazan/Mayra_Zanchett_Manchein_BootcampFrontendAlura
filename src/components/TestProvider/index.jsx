import React from 'react';
import { ThemeProvider } from 'styled-components';
import PropTypes from 'prop-types';
import theme from '../../theme';
import GlobalStyle from '../../theme/GlobalStyle';

const TestProvider = ({ children }) => (
  <ThemeProvider theme={theme}>
    <GlobalStyle />
    {children}
  </ThemeProvider>
);

TestProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default TestProvider;