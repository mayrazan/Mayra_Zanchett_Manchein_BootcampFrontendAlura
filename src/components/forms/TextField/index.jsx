/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const InputWrapper = styled.div`
  margin-bottom: 17px;
`;

const Input = styled.input`
  width: 100%;
  border: 1px solid ${({ theme }) => theme.colors.secondary.color};
  padding: 12px 16px;
  outline: 0;
  border-radius: 15px;
  box-shadow: 0 10px 15px 0 rgba(111,30,81,0.05);
`;

export default function TextField({
  placeholder,
  name,
  onChange,
  value,
  as,
  ...props
}) {
  return (
    <InputWrapper>
      <Input
        placeholder={placeholder}
        name={name}
        onChange={onChange}
        value={value}
        required
        as={as}
        {...props}
      />
    </InputWrapper>
  );
}

TextField.propTypes = {
  placeholder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  as: PropTypes.string,
};

TextField.defaultProps = {
  as: 'input',
};
