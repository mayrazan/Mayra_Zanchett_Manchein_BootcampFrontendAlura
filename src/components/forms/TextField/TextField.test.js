import React from 'react';
import user from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import TextField from '.';
import TestProvider from '../../TestProvider';

describe('<TextField />', () => {
  test('renderiza componente', () => {
    render(
      <TestProvider>
        <TextField
          placeholder="Nome"
          value="Mayra"
          onChange={() => {}}
          name="nome"
        />
      </TestProvider>,
    );
    const textField = screen.getByPlaceholderText(/nome/i);
    expect(textField).toMatchSnapshot();
  });

  describe('when field is valid', () => {
    describe('and user is typing', () => {
      test('the value must be updated', () => {
        const onChangeMock = jest.fn();
        render(
          <TestProvider>
            <TextField
              placeholder="Nome"
              value="Mayra"
              onChange={onChangeMock}
              name="nome"
            />
          </TestProvider>,
        );

        const inputNome = screen.getByPlaceholderText(/nome/i);
        user.type(inputNome, 'mayra');
        expect(onChangeMock).toHaveBeenCalledTimes(5);
      });
    });
  });
});
