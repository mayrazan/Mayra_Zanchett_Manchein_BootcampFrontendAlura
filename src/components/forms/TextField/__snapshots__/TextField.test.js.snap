// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> renderiza componente 1`] = `
.c0 {
  width: 100%;
  border: 1px solid rgb(111,30,81);
  padding: 12px 16px;
  outline: 0;
  border-radius: 15px;
  box-shadow: 0 10px 15px 0 rgba(111,30,81,0.05);
}

<input
  class="c0"
  name="nome"
  placeholder="Nome"
  required=""
  value="Mayra"
/>
`;
